src/kontrol/kdist/cheatcodes.md

```diff
@@ -435,7 +435,28 @@ This rule returns a symbolic boolean value being either 0 (false) or 1 (true).
          <k> #call_foundry SELECTOR _ => . ... </k>
          <output> _ => #buf(32, ?WORD) </output>
       requires SELECTOR ==Int selector ( "freshBool()" )
-       ensures #rangeBool(?WORD)
+       ensures #rangeBool(?WORD2)
+       [preserves-definedness]
+```
+
+#### `freshBytes` - Returns a single symbolic boolean.
+
+```
+function freshBytes(uint256) external returns (bytes memory);
+```
+
+`foundry.call.freshBytes` will match when the `freshBytes` cheat code function is called.
+This rule returns a fully symbolic byte array value of the given length.
+
+```{.k .symbolic}
+    rule [foundry.call.freshBytes]:
+         <k> #call_foundry SELECTOR ARGS => . ... </k>
+         <output> _ => 
+            #buf(32, 32) +Bytes #buf(32, #asWord(ARGS)) +Bytes ?BYTES 
+            +Bytes #buf ( ( ( notMaxUInt5 &Int ( #asWord(ARGS) +Int maxUInt5 ) ) -Int #asWord(ARGS) ) , 0 ) 
+         </output>
+      requires SELECTOR ==Int selector ( "freshBytes(uint256)" )
+      ensures lengthBytes(?BYTES) ==Int #asWord(ARGS)
        [preserves-definedness]
```

Comment:
```suggestion
      [preserves-definedness]
```

Reply:
This is just for alignment with `ensures` in the previous line.

Reply:
Fixed.